runtime/interpreter/statictype.go

```diff
@@ -255,6 +256,21 @@ func NewVariableSizedStaticType(
 	}
 }
 
+func (t VariableSizedStaticType) IsComposite() bool {
+	return false
+}
+
+func (t VariableSizedStaticType) Copy() atree.TypeInfo {
+	// TODO: is creating a new VariableSizedStaticType necessary?
+	// atree needs immutable TypeInfo or a copy of mutable TypeInfo because TypeInfo can be shared.
+	return t
+}
+
+func (t VariableSizedStaticType) Identifier() string {
+	// TODO: is ID guaranteed to be unique?
+	return string(t.ID())
```

Comment:
What is the uniqueness requirement of `Identifier()`? All types that have the same "contents" have the same ID. Is that sufficient? Or does "unique" mean here that every type (here, `VariableSizedStaticType`) gets a unique ID, even for other types that are "equal"?

Reply:
> All types that have the same "contents" have the same ID. Is that sufficient? Or does "unique" mean here that every type (here, VariableSizedStaticType) gets a unique ID, even for other types that are "equal"?

Different types must have different identifiers.  For example, `VariableSizedStaticType` and `ConstantSizedStaticType` must have different identifiers, even though "contents" are the same. This is used by atree for deduplication during encoding.